### ISSUE TITLE
Performance issue in snlds/examples/lorenz_attractor_trainer.py

### ISSUE THREAD

**Issue by DLPerf** [yerkojahve/Meta-Pseudo-label#1]
Hello! Our static bug checker has found a performance issue in snlds/examples/lorenz_attractor_trainer.py:  Python type arguments  `learning_rate`, `temperature` and `cross_entrophy_coef` [(here)](https://github.com/yerkojahve/Meta-Pseudo-label/blob/44a7cbc6429505f6e54fee432a78be4a84c9d144/snlds/examples/lorenz_attractor_trainer.py#L412) are passed to tf.function decorated function [`train_step`](https://github.com/yerkojahve/Meta-Pseudo-label/blob/44a7cbc6429505f6e54fee432a78be4a84c9d144/snlds/examples/lorenz_attractor_trainer.py#L188) instead of tensors. As there is a [while loop](https://github.com/yerkojahve/Meta-Pseudo-label/blob/44a7cbc6429505f6e54fee432a78be4a84c9d144/snlds/examples/lorenz_attractor_trainer.py#L406), every time the value changes, a new graph will be generated, and that can trigger tf.function retracing warning.

Here is [the tensorflow document](https://tensorflow.google.cn/guide/function#pass_tensors_instead_of_python_literals) to support it.

Looking forward to your reply.

**Comment by DLPerf** [comment]
We are investigating this kind of issues, and your answer will be of great help to our work. Can you take a look? Maybe pass them as tf.constant type values. Thank you in advance! @jiawen @cghawthorne @jaingaurav @dfurrer
